### ISSUE TITLE
Using gnomAD-SV and Clingen-SV

### ISSUE THREAD

**Issue by prasundutta87** [lgmgeo/AnnotSV#59]
Hi,

I am dealing with some SVs that I detected using long reads (Oxford Nanopore). In order to prioritise variants, I was using VEP and SnpEff to add functional annotation to my SV sets. I had also come across AnnotSV and it was also recommended in one of my questions in Biostars and hence I am here.

Although, I was successful in installing the software, I was wondering how can I use custom annotations such as gnomAD-SV (https://www.ncbi.nlm.nih.gov/dbvar/studies/nstd166/) and ClinVar-SV (https://www.ncbi.nlm.nih.gov/dbvar/studies/nstd102/). I am using GRCh38 and one of my aims for variant prioritisation is to remove common variants in my population of interest. I get population based SV frequencies from gnomAD-SV. How can I use AnnotSV to achieve this?

Also, in the readme file, it is written: **To update the data sources, please download the latest available files (not the ones given 
as an example in the README).**

I used this command to download the human annotation: make PREFIX=... install-human-annotation

Does it not download whatever is latest?

Regards,
Prasun Dutta

**Comment by prasundutta87** [comment]
Hi Again,

I think that I have figured out the process of adding custom annotation and also do the population based AF filtering from the output tsv file. However, I was not able to understand the difference between “FtIncludedInSV” directory and  “SVincludedInFt” directory. I would be grateful if you could kindly let me know the difference as the readme was not very clear to me and where exactly should I put my bed file of interest. 

Currently, for testing I have put in both places, however I am getting two columns of my AF values. Not sure why. Probably because gnomAD-SV has multiple variant rows with same start and end coordinates due to the presence of multi-allelic CNVs (mCNVs)? Have you encontered this kind of issue before?

Regards,
Prasun

**Comment by lgmgeo** [comment]
Hi Prasun,

Glad to read you've figured out the process of adding custom annotations.
Just in case, to help potential future readers of this post, this is explained in the "d) External BED annotation files (optional)" section of the [README](https://www.lbgi.fr/AnnotSV/Documentation/README.AnnotSV_latest.pdf).

Regarding the difference between “FtIncludedInSV” directory and “SVincludedInFt” directory:
- By placing your BED file in the “FtIncludedInSV” directory, only the features (e.g. pathogenic regions) **overlapped at 100% with the SV** will be reported.
- By placing your BED file in the “SVincludedInFt” directory, only the features (e.g. benign regions) **overlapping 100% of the SV** will be reported. 

So, in my opinion, your gnomAD-SV BED file should be placed in the “SVincludedInFt” directory.
Let me explain:
You have genomic regions from gnomAD. Frequent gnomAD SV (obtained after your population-based AF filtering) correspond to benign regions. This "benign SV" annotation might only be useful if overlapping 100% of your "SV to annotate". I mean, only if your "SV to annotate" is completely contained within that benign genomic region. Otherwise, if the "SV to annotate" includes additional genomic material (compared to the benign SV), you can not conclude on its benign/pathogenic status. 

You may be interested in the answers of the following questions in the README FAQ:
- Q: How overlaps (%) are calculated? 
- Q: Why not to use a reciprocal overlap with features overlapped with the SV to annotate?

Best regards,
Véronique

**Comment by lgmgeo** [comment]
> Also, in the readme file, it is written: To update the data sources, please download the latest available files (not the ones given
as an example in the README).

Please, to update the data sources on your own, look at the "Updating the data source" section in the README from each annotation source ("Gene annotations", "Regulatory Elements annotations"...).
It's time consuming and I achieve a new version with updated annotations only once to twice a year.

> I used this command to download the human annotation: make PREFIX=... install-human-annotation
Does it not download whatever is latest?

Actually no. It downloads the latest annotations that I provide with AnnotSV, but not the latest from each data source

**Comment by prasundutta87** [comment]
Hi @lgmgeo ,

Thank you so much for your replies and explaination.

Regarding gnomAD-SV data, I just processed the https://ftp.ncbi.nlm.nih.gov/pub/dbVar/data/Homo_sapiens/by_study/vcf/nstd166.GRCh38.variant_call.vcf.gz dataset. I used SnpSift to take out the chrom, start, end and the AF data and converted the coordinates to 0-based system to get my bed file. I did not do any filtering at all. Using AnnotSV allowed me to get the AF values associated with my SVs and then I tried doing my filtering after that. Please let me know if it is the correct way to do that, or I should filter my bed file to only keep benign SVs (population AF >0.05) and then just remove those SV which get annotated by the AF values?

However, I am getting values like this in the output: 0;2.62E-04

This may be because of multi-allelic CNVs (mCNVs). Is there a way to resolve that, or I need to process it further (probably using R)?

Regards,
Prasun

**Comment by lgmgeo** [comment]
Hi,

There are pros and cons for each method you offer. 

> My plan was to annotate my SVs with AnnotSV to get the AF values associated with the SVs and then perform filtering (mostly using excel). 

This way, you will have to filter your data for each analysis (repetitive). But you keep the possibility to change the threshold (AF> 0.05) at any time.

> or I should filter my bed file to only keep SVs common to my population of SVs (can be benign) (population AF >0.05) and then just remove those SV which get annotated by the AF values? 

This way, you will only need to filter the BED file once. But you couldn't change the threshold.

Each method makes sense to me. If your threshold is fixed, the second method is best. Otherwise, please choose the first one.

> Furthermore, I am getting values like this in the output: 0;2.62E-04

It's normal.
Indeed, it could be due to mCNV (same start and same stop).
But not only. This may be because your SV to annotate is completely overlapped with 2 benign SVs (one with AF = 0, the other with AF = 2.62E-04) with different coordinates.
You cannot therefore remove these multiple AFs from your annotations.

2 solutions:
- If you have a fixed threshold AF value, filter your bed file to keep only the common SVs (considered benign). And then just remove the SV which get annotated by the AF values.
- Otherwise, process your data to keep only the minimum AF frequency. Thus, you will be able to filter your data using excel.

Best,

Véronique

**Comment by prasundutta87** [comment]
Thank you so much for this. I will try to perform the annotation again now and will come back here if I have any questions.

**Comment by prasundutta87** [comment]
Thanks @lgmgeo for the reply. It was not very clear to me that CLN stands for Clinvar. I have checked my output file now and some of my hits have got that information. Sorry, if my second question was not clear. I am aware of the VCF file used and I had found it in the README and that is the full Clinvar dataset. The link I shared above are structural variants with clinical assertions, submitted to ClinVar by external labs. I guess there may be some overlap (I haven't checked personally) and may be added as a custom dataset.

Thanks for letting me know that a new update is coming.
